kolibri/core/tasks/utils.py

```diff
@@ -122,12 +126,52 @@ def shutdown(self):
         self.stop()
 
 
-class DiskCacheRLock(RLock):
+class DiskCacheRLock(object):
+    """
+    Vendored from
+    https://github.com/grantjenks/python-diskcache/blob/2d1f43ea2be4c82a430d245de6260c3e18059ba1/diskcache/recipes.py
+    """
+
+    def __init__(self, cache, key, expire=None):
+        self._cache = cache
+        self._key = key
+        self._expire = expire
+
+    def acquire(self):
+        "Acquire lock by incrementing count using spin-lock algorithm."
+        pid = os.getpid()
+        tid = get_ident()
+        pid_tid = "{}-{}".format(pid, tid)
+
+        while True:
+            value, count = self._cache.get(self._key, (None, 0))
+            if pid_tid == value or count == 0:
+                self._cache.set(
+                    self._key, (pid_tid, count + 1), self._expire,
+                )
+                return
+            time.sleep(0.001)
+
     def release(self):
-        super(DiskCacheRLock, self).release()
+        "Release lock by decrementing count."
+        pid = os.getpid()
+        tid = get_ident()
+        pid_tid = "{}-{}".format(pid, tid)
+
+        value, count = self._cache.get(self._key, default=(None, 0))
+        is_owned = pid_tid == value and count > 0
+        assert is_owned, "cannot release un-acquired lock"
+        self._cache.set(self._key, (value, count - 1), self._expire)
+
         # RLOCK leaves the db connection open after releasing the lock
         # Let's ensure it's correctly closed
         self._cache.close()
```

Comment:
If I understand this correctly, the lock is a key entry into the Django cache, that would be usually either redis or memory. If that's the case we don't need to close this , as there's not database involved.